### PR TITLE
Allow custom IDs

### LINKED ISSUE #25: Creating a Model with a custom id is no longer possible
Test case:

```
class MyModel extends Model<string> {}

await MyModel.create({
  _id: 'test'
});
```

This doesn't create anything in the database and returns an instance of `MyModel` without any error. Calling `.save()` afterwards also doesn't save it.

### PR DESCRIPTION
**Pull request by Xstoudi**
Should resolve #25

### CODE REVIEW

**Comment by stropitek** [comment]
There is an issue in the diff. All the lines of the files you modified have changed.

File: src/Model/__tests__/Model.test.ts [hunk 1]
```diff
@@ -1,294 +1,307 @@
...
+  expect(user.username).toBeDefined();
+  expect(user.createdAt).toBe(createdAt);
+});
+
+test('pass custom id', async () => {
+  const user = await User.create({
+    _id: 'test',
+    username: nextUsername(),
+    password: 'mypass',
+  } as any);
```
Comment (targos) [inline src/Model/__tests__/Model.test.ts:301]: why do you need `any` here?

**Comment by Xstoudi** [comment]
I fixed the type to allow you to pass _id without passing any, tell me if you consider it isn't the best way to do it.

For the line diff issue, I fixed it too.

**Comment by targos** [comment]
For `_id`, we can just define it on the Model directly (public readonly _id) like in the public types: https://github.com/zakodium/adonis-mongodb/blob/210cacdd1f61d512bf58155578c7a21fa85d7f3c/src/types.ts#L149

**Comment by Xstoudi** [comment]
It's already the case in the types.ts so it's only tests problem.
I added the field as any, as there is no type-concern in the Model.

**Review by targos — APPROVED** [review]

**Pull request merged**